recipes/apr/all/conanfile.py

```diff
@@ -0,0 +1,110 @@
+    }
+
+    _autotools = None
+    _cmake = None
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        del self.settings.compiler.cppstd
+        del self.settings.compiler.libcxx
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("{}-{}".format(self.name, self.version), self._source_subfolder)
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.definitions["INSTALL_PDB"] = False
+        self._cmake.definitions["APR_BUILD_TESTAPR"] = False
+        # self._cmake.definitions["APR_INSTALL_PRIVATE_H"] = True
+        self._cmake.configure(build_folder=self._build_subfolder)
+        return self._cmake
+
+    def _configure_autotools(self):
+        if self._autotools:
+            return self._autotools
+        self._autotools = AutoToolsBuildEnvironment(self)
+        conf_args = [
+            "--with-installbuilddir=${prefix}/bin/build-1",
+        ]
+        if self.options.shared:
+            conf_args.extend(["--enable-shared", "--disable-static"])
+        else:
+            conf_args.extend(["--disable-shared", "--enable-static"])
+        self._autotools.configure(args=conf_args, configure_dir=self._source_subfolder)
+        return self._autotools
+
+    def _patch_sources(self):
+        for patch in self.conan_data["patches"][self.version]:
+            tools.patch(**patch)
+
+    def build(self):
+        self._patch_sources()
+        if self.settings.os == "Windows":
+            cmake = self._configure_cmake()
+            cmake.build(target="libapr-1" if self.options.shared else "apr-1")
+        else:
+            autotools = self._configure_autotools()
+            autotools.make()
+
+    def package(self):
+        self.copy("LICENSE", dst="licenses", src=self._source_subfolder)
+        if self.settings.os == "Windows":
+            cmake = self._configure_cmake()
+            cmake.install()
+        else:
+            autotools = self._configure_autotools()
+            autotools.install()
+
+            os.unlink(os.path.join(self.package_folder, "lib", "libapr-1.la"))
+            tools.rmdir(os.path.join(self.package_folder, "build-1"))
+            tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+
+    def package_info(self):
+        self.cpp_info.names["pkg_config"] = "apr-1"
+        self.cpp_info.libs = ["apr-1"]
+        if not self.options.shared:
+            self.cpp_info.defines = ["APR_DECLARE_STATIC"]
+            if self.settings.os == "Linux":
+                self.cpp_info.system_libs = ["dl", "pthread"]
```

Comment:
```suggestion
        if self.settings.os == "Linux":
            self.cpp_info.system_libs = ["dl", "pthread"]
```

Reply:
This is correct.
Shared libraries don't need any extra libraries.
The headers also do not expose any symbols other then the `apr_*` ones.

Reply:
okay, fine for this case, but remember we already discussed about it few months ago. checking for it increases the recipes complexity, also exposing those libraries are harmless and the user will need them anyway.